fuzz/fuzz_targets/encode_decode.rs

```diff
@@ -33,12 +33,17 @@ fn do_test(data: &[u8]) {
             match Hrp::parse(&s) {
                 Err(_) => return,
                 Ok(hrp) => {
-                    if let Ok(data_str) = bech32::encode(hrp, &dp, variant).map(|b32| b32.to_string()) {
-                        let decoded = bech32::decode(&data_str);
-                        let b32 = decoded.expect("should be able to decode own encoding");
+                    let encoded = bech32::encode(hrp, &dp, variant).expect("failed to encode");
```

Comment:
In e2527dc19c3e8e9a4cd678a955fc5b161e175c87:

Should we assert that the encoding here matches the original string? At least, up to the length of the encoded data?

Reply:
We do that in `decode_rnd` although I'm not totally sure why we have both - do you see benefit in both of have I changed them so much that some previous benefit is gone? This test attempts to help the fuzzer find good addresses but with my new-found belief in the wonder magic of the fuzzer I'm not sure how useful that is?

Reply:
Yes, there is value in both. One is checking that an arbitrary string can be decoded and re-encoded. The other is checking that an arbitarry hrpstring can be encoded and re-decoded.